File: problemas/strings/2587/2587.lua
function letra(palavra, pos)
    return string.sub(palavra, pos, pos)
end

c = tonumber(io.read())

for _ = 1, c do
    palavra1 = io.read()
    palavra2 = io.read()
    secreta = io.read()

    pos1 = string.find(secreta, '_')
    pos2 = string.find(secreta, '_', pos1 + 1)

    if letra(palavra1, pos1) == letra(palavra2, pos2) or letra(palavra1, pos2) == letra(palavra2, pos1) then
        print("Y")
    else
        print("N")
    end
end

File: problemas/strings/2587/2587.ml
Scanf.scanf "%d\n" (fun n ->
	for i = 1 to n do
		Scanf.scanf "%s\n%s\n%s\n" (fun palavra1 palavra2 secreta ->
			let pos1 = String.index_from secreta 0 '_' in
			let pos2 = String.index_from secreta (pos1 + 1) '_' in
			if (palavra1.[pos1] == palavra2.[pos2] || palavra2.[pos1] == palavra1.[pos2]) then
				Printf.printf "Y\n"
			else
				Printf.printf "N\n")
	done;
)
